test/test_autograd.py

```diff
@@ -611,6 +611,19 @@ def backward(ctx, grad_b):
 
         TestFn.apply(b).sum().backward()
 
+    def test_free_deep_graph(self):
+        def scope():
+            depth = 150000
+            x = torch.randn(9, requires_grad=True)
```

Comment:
:O not sure if this is the best way to test, but I can't think of better ones. Can we change it to a numel-1 tensor?

Reply:
Well, it consistently segfaults :D

But yes taking suggestions on better tests...

Reply:
Why is that a better test? The chain is an easy way to get a segfault.

Reply:
Yes, but a general dag tests the logic better. You can augment the chain to make it a bit more complex. Anyways, not that important.

Reply:
Makes sense! I'll add in something that looks less like the chain soon